Description: Build fixes for Ubuntu Kinetic
Author: Luciano Lo Giudice <luciano.logiudice@canonical.com>
Forwarded: no

---
diff --git a/src/include/buffer.h b/src/include/buffer.h
index 0c89367dd3..f8fd9573f0 100644
--- a/src/include/buffer.h
+++ b/src/include/buffer.h
@@ -49,6 +49,7 @@
 
 #include <exception>
 #include <type_traits>
+#include <memory>
 
 #include "page.h"
 #include "crc32c.h"
diff --git a/src/spdk/lib/iscsi/iscsi.c b/src/spdk/lib/iscsi/iscsi.c
index febf4cac48..88cb41a52e 100644
--- a/src/spdk/lib/iscsi/iscsi.c
+++ b/src/spdk/lib/iscsi/iscsi.c
@@ -62,7 +62,7 @@
 #define SPDK_CRC32C_INITIAL    0xffffffffUL
 #define SPDK_CRC32C_XOR        0xffffffffUL
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || __has_include(<bsd/stdlib.h>)
 #define HAVE_SRANDOMDEV 1
 #define HAVE_ARC4RANDOM 1
 #endif
